charts/gardener/gardenlet/charts/runtime/templates/clusterrole-gardenlet.yaml

```diff
@@ -0,0 +1,293 @@
+{{- if .Values.global.gardenlet.enabled }}
+# ClusterRole defines the required permissions for the gardenlet in the seed
+# For now, we provide escalation privileges required by gardener resoruce manager.
+---
+apiVersion: {{ include "rbacversion" . }}
+kind: ClusterRole
+metadata:
+  name: gardener.cloud:system:gardenlet
+  labels:
+    gardener.cloud/role: gardenlet
+    app: gardener
+    role: gardenlet
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - list
+  - watch
+  - delete
+  - deletecollection
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  - namespaces
+  - secrets
+  - serviceaccounts
+  - services
+  verbs:
+  - create
+  - delete
+  - deletecollection
+  - get
+  - list
+  - watch
+  - patch
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - persistentvolumeclaims
+  verbs:
+  - get
+  - list
+  - watch
+  - patch
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - persistentvolumeclaims
+  resourceNames:
+  - alertmanager-db-alertmanager-0
+  - loki-loki-0
+  - prometheus-db-prometheus-0
+  verbs:
+  - delete
+- apiGroups:
+  - admissionregistration.k8s.io
+  resources:
+  - mutatingwebhookconfigurations
+  - validatingwebhookconfigurations
+  verbs:
+  - create
+- apiGroups:
+  - admissionregistration.k8s.io
+  resources:
+  - mutatingwebhookconfigurations
+  resourceNames:
+  - vpa-webhook-config-seed
+  verbs:
+  - get
+  - delete
+  - update
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - create
+  - get
+  - list
+  - watch
+  - patch
+  - update
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  resourceNames:
+  - hvpas.autoscaling.k8s.io
+  verbs:
+  - delete
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  - statefulsets
+  - replicasets
```

Comment:
The case described in https://github.com/gardener/gardener/pull/3763#discussion_r602357037 still fails:

```
time="2021-04-09T13:37:03+03:00" level=error msg="Failed to reconcile Shoot \"aws-local\": flow \"Shoot cluster reconciliation\" encountered task errors: [task \"Waiting until Kubernetes API server reports readiness\" failed: failure to read the logs: pods \"kube-apiserver-7968db85fd-gtd45\" is forbidden: User \"system:serviceaccount:garden:gardenlet\" cannot get resource \"pods/log\" in API group \"\" in the namespace \"shoot--foo--aws-local\": retry failed with context deadline exceeded, last error: kube-apiserver deployment has outdated replicas]" operation=reconcile shoot=garden-foo/aws-local
```